### PR TITLE
Real-time status about sync details

### PR DESCRIPTION
**Pull request by boatbomber**
Resolves https://github.com/rojo-rbx/rojo/issues/418.

image

Features:
- Human-readable timestamp text (just now, 3 seconds ago, 1 minute ago, 2 minute**s** ago)
- How many changes were applied by the latest patch
- Timestamp text live updates (and slows the update freq as it gets older)
- Combines patches that are within a second of each other, to more readably handle burst changes

### CODE REVIEW

**Comment by OverHash** [comment]
Do users need to know how many changes have occurred in a given time period? I feel like just displaying "last synced just now" or "last synced 5 minutes ago" is enough.

**Comment by boatbomber** [comment]
> Do users need to know how many changes have occurred in a given time period? I feel like just displaying "last synced just now" or "last synced 5 minutes ago" is enough.

@OverHash  The #418 issue specifically asked to know a count, and no one disagreed. That said, I do think that a simple last since readout would solve the most common use case put forth in the issue- it informs you that the plugin is still working. However, it's possible that there could be cases where it's only syncing partially, which wouldn't be exposed by just a timestamp.

File: plugin/src/App/init.lua [hunk 1]
```diff
@@ -123,9 +124,15 @@ function App:startSession()
 		end
 
 		if count == 0 then return end
+
+		local old = self.patchInfo:getValue()
+		if now - old.timestamp < 2 then
+            count += old.updates
```
Comment (Kampfkarren) [inline plugin/src/App/init.lua:130]: Weird indentation

Comment (boatbomber) [reply plugin/src/App/init.lua:130]: Wrote it on mobile, sorry! I'll fix it when I get home later tonight.

**Review by Logic-Mastermind — APPROVED** [review]
I see nothing that could be improved on in terms of code, I think this should be merged as soon as possible.

File: plugin/src/App/StatusPages/Connected.lua [hunk 2]
```diff
@@ -13,6 +13,26 @@ local BorderedContainer = require(Plugin.App.Components.BorderedContainer)
 
 local e = Roact.createElement
 
+local AGE_UNITS = { {31556909, "year"}, {2629743, "month"}, {604800, "week"}, {86400, "day"}, {3600, "hour"}, {60, "minute"}, }
```
Comment (LPGhatguy) [inline plugin/src/App/StatusPages/Connected.lua:16]: "last synced one year ago" would be a pretty funny message to happen in practice

**Review by LPGhatguy — APPROVED** [review]
Very clean, thank you!

**Pull request merged**